File: f77/build06/def_data.f
C
      Subroutine Default_Data
C
C     This subroutine sets the default data values.
C
C     Inputs:
C       None.
C
C     Outputs:
C       DtMax                   - Maximum allowable simulation time step [sec]
C       DtMin                   - Minimum allowable simulation time step [sec]
C       DtPrint                 - Time step between printing data [sec]
C       Guidance_Gain   - Commanded angular rate per measured LOS rate [Real]
C       RDTODG                  - Radians to degrees conversion factor [deg/rad]
C       Time0                   - Initial time [sec]
C       Theta_IC_dg             - Initial attitude of missile [deg]
C       Tstop                   - Simulation stop time [sec]
C       Velocity                - Velocity of missile [m/sec]
C       X_IC                    - Initial X position of missile [m]
C       Z_IC                    - Initial Z position of missile [m]
C
C     Internal variables and constants
C       None.
C
C     Declare global common
C
      Include '../driver/global.inc'
      Include '../driver/sysvars.inc'
      Equivalence (Real_Array(10), RDTODG)
      Equivalence (Real_Array(12), Theta_IC_dg)
      Equivalence (Real_Array(13), Velocity)
      Equivalence (Real_Array(14), X_IC)
      Equivalence (Real_Array(15), Z_IC)
      Equivalence (Real_Array(25), Guidance_Gain)
C
C     Set model independent data and reset time
C

      DtMax   =  0.005e0
      DtMin   = 0.001e0
      DtPrint =  0.01e0
      Time0   =  0.0e0
      Tstop   = 10.0e0
      RDTODG  = 180.0e0/(4.0e0*atan(1.0e0))
C
C     Set the default data for the simulation
C
      Theta_IC_dg   =    0.0e0
      Velocity      =  100.0e0
      X_IC          = -500.0e0
      Z_IC          = -100.0e0
      Guidance_Gain =    3.0e0
C
      Return
      End

File: gcc/build06/def_data.c
#include "def_data.h"
#include "../driver/global.h"
#include "../driver/sysvars.h"
#include <math.h>

#define rdtodg         real_array[10].r  // Radians to degrees conversion factor [deg/rad]
#define theta_ic_dg    real_array[12].r  // Initial attitude of missile [deg]
#define velocity       real_array[13].r  // Velocity of missile [m/sec]
#define x_ic           real_array[14].r  // Initial X position of missile [m]
#define z_ic           real_array[15].r  // Initial Z position of missile [m]
#define guidance_gain  real_array[25].r  // Commanded angular rate per measured LOS rate [Real]

void default_data(void) {
  // Set model independent data and reset time
  dtmax   = 0.005;
  dtmin   = 0.001;
  dtprint = 0.01;
  time0   = 0.0;
  tstop   = 10.0;
  rdtodg  = 180.0/M_PI;
  // Set the default data for the simulation
  theta_ic_dg = 0.0;
  velocity    = 100.0;
  x_ic        = -500.0;
  z_ic        = -100.0;
  guidance_gain = 3.0;
}
